ingestion/src/metadata/profiler/source/metadata_ext.py

```diff
@@ -225,18 +235,38 @@ def filter_entities(self, tables: Iterable[Table]) -> Iterable[Table]:
         """
         for table in tables:
             try:
+                schema_fqn = fqn.build(
+                    self.metadata,
+                    entity_type=DatabaseSchema,
+                    service_name=self.config.source.serviceName,
+                    database_name=table.database.name,
+                    schema_name=table.databaseSchema.name,
```

Comment:
Shouldn't this be constructed from `self.source_config.schemaFilterPattern`? Same thing with the table but from `self.source_config.tableFilterPattern` and database but from `self.source_config.databaseFilterPattern`?

Reply:
From what I am understand -- maybe I am missing something -- if you have tables ["tableA", "tableB",..."tableN"] where schema is "foo" and database "bar".

if you set your tableFilterPattern as `bazz.foobar.tableA` and set `useFqnForFiltering` to `True`. 

When you iterate over tables you will build you fqn in `fqn.build()` as `bar.foo.tableA`, since we are using `table` to get database, schema and table name. In this scenario you would effectively process `foo.bar.tableA` when we want to process `bazz.foobar.tableA`.

Let me know if that makes sense.

Reply:
The filter will already have an fqn that is passed from the config
we need to create the fqn in order to compare it with the filterPattern.